uber/site_sections/registration.py

```diff
@@ -888,6 +888,63 @@ def discount(self, session, message='', **params):
 
         return {'message': message}
 
+    @csv_file
```

Comment:
This was all hella cleaned up from the original feature branch. Some functionality didn't make it, particularly the ability to email promo codes. There wasn't a well-defined-enough use-case, and while I initially intended to let folks generate a 'template email' so they could copy-paste it, I looked into what it takes to do that and I couldn't really justify spending that much time on it. In any case, this stuff is pretty solid.

FWIW, I really, really want to move this stuff into `/budget.py` (and the companion templates too, of course). The fact is, with this feature you can create an unlimited-use code that makes everyone's badge free and just hand it out. That should be SUPER restricted. I left it here for this commit because I wanted to have a useable diff between the original feature branch and my changes.